homeassistant/components/google_travel_time/sensor.py

```diff
@@ -127,31 +128,20 @@ async def async_setup_entry(
             config_entry, data=new_data, options=options
         )
 
-    def run_setup(event):
-        """
-        Delay the setup until Home Assistant is fully initialized.
-
-        This allows any entities to be created already
-        """
-        hass.data.setdefault(DATA_KEY, [])
+    hass.data.setdefault(DATA_KEY, [])
 
-        api_key = config_entry.data.get(CONF_API_KEY)
-        origin = config_entry.data.get(CONF_ORIGIN)
-        destination = config_entry.data.get(CONF_DESTINATION)
-        name = config_entry.data.get(
-            CONF_NAME, f"{DEFAULT_NAME}: {origin} -> {destination}"
-        )
-
-        sensor = GoogleTravelTimeSensor(
-            hass, config_entry, name, api_key, origin, destination
-        )
-        hass.data[DATA_KEY].append(sensor)
+    api_key = config_entry.data.get(CONF_API_KEY)
+    origin = config_entry.data.get(CONF_ORIGIN)
+    destination = config_entry.data.get(CONF_DESTINATION)
+    name = name or f"{DEFAULT_NAME}: {origin} -> {destination}"
 
-        if sensor.valid_api_connection:
-            async_add_entities([sensor], True)
+    sensor = GoogleTravelTimeSensor(
+        hass, config_entry, name, api_key, origin, destination
+    )
+    hass.data[DATA_KEY].append(sensor)
 
-    # Wait until start event is sent to load this component.
-    await hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+    if sensor.valid_api_connection:
```

Comment:
A future improvement would be to start a reauth flow of the credentials are bad.

Reply:
Like the Waze one we could also add some validation during the config flow which I felt was out of scope for this PR